hi-ml-cpath/src/health_cpath/utils/viz_utils.py

```diff
@@ -196,16 +204,34 @@ def plot_heatmap_overlay(
                                          location_bbox=slide_dict[SlideKey.ORIGIN],
                                          scale_factor=slide_dict[SlideKey.SCALE],
                                          should_upscale_coords=should_upscale_coords)
-    cmap = plt.cm.get_cmap("Reds")
+    cmap = plt.cm.get_cmap("Spectral_r")  # _r reverse the color map so that the highest attention is red
 
     tile_xs, tile_ys = sel_coords.T
     rects = [patches.Rectangle(xy, tile_size, tile_size) for xy in zip(tile_xs, tile_ys)]
 
-    pc = collection.PatchCollection(rects, match_original=True, cmap=cmap, alpha=0.5, edgecolor="black")
+    # line width is set to 0 to avoid the black border around the tiles as the tiles are already colored
+    pc = collection.PatchCollection(rects, match_original=True, cmap=cmap, alpha=0.5, linewidth=0)
     pc.set_array(np.array(attentions))
-    pc.set_clim([0, 1])
-    ax.add_collection(pc)
-    plt.colorbar(pc, ax=ax)
+    ax1.add_collection(pc)
+    plt.colorbar(pc, cax=cax)  # add colorbar to the right of the plot (cax)
```

Comment:
the position of the colorbar is very far to the right in some examples. can this be moved closer to the image?

Reply:
done